### DISCUSSION TITLE
How do I use my CSS classes and tailwind in RadixUi?

### DISCUSSION THREAD

**Discussion by mtalha218** [radix-ui/themes#148] [Q&A]
I am having some issues of using tailwindcss and CSS classes in RadixUI, 

`<Button className="rounded-none text3">Upload Documents</Button>`

Now neither my CSS class text3 is applied, nor did tailwindCss class is applied. Why is that? Am i implementing it wrong?

**Comment by vladmoroz** [comment]
Please post a reproduction on CodeSandbox that we could look at.

Although, overall I wouldn't recommend to customize the components this way instead of using the provided props, as the code will get unmaintainable and the design incoherent quite fast.

You can use size and radius prop instead:
```jsx
<Button size="3" radius="none">Upload Documents</Button>
```

**Reply by mtalha218** [reply]
So customizing RadixUi components is quiet difficult, suppose if the requirement is very different from the original component.

**Reply by vladmoroz** [reply]
Technically most components work with Tailwind if your setup is right, it’s just my recommendation when it comes to using the Radix Themes to get the best experience out of it.

**Reply by mtalha218** [reply]
Also not even my css class not working

`<Button className="text3">Upload Documents</Button>`

There must be an explanation to that.

**Reply by mtalha218** [reply]
I have figured something out, i am unable to use className as it is not being override on radixUi styles. This works for me.

`<TextField.Input
                className="[&&]:h-[57px] [&]:rounded-[8px] text3 [&&]:border-1 [&&]:border-solid [&&]:border-[#FFFFFF] "
                placeholder="Enter your email"
              />`

But i have to add "[&&]:" to make that class a priority, can i fix that so that i dont have to use "[&&]:" every time and className should override the radixUi styles?

**Reply by vladmoroz** [reply]
See https://github.com/radix-ui/themes/issues/109#issuecomment-1747345743 for conflict-free setup with Tailwind
